cmd/show.go

```diff
@@ -0,0 +1,26 @@
+package cmd
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/driusan/dgit/git"
+)
+
+func Show(c *git.Client, args []string) error {
+	flags := flag.NewFlagSet("show", flag.ExitOnError)
+	flags.SetOutput(flag.CommandLine.Output())
+	flags.Usage = func() {
+		flag.Usage()
+		fmt.Fprintf(flag.CommandLine.Output(), "\n\nOptions:\n")
+		flags.PrintDefaults()
+	}
+
+	opts := git.ShowOptions{}
+	flags.Var(&opts.Format, "format", "Print the contents of commit logs in a specified format")
+	flags.Var(&opts.Format, "pretty", "Alias for --format")
```

Comment:
Is this (multiple flags going to the same place) valid or do they overwrite each other?

Reply:
They go to the same place. If there is a collision then an error is produced. See git/show.go FormatString type.

Reply:
How does the error propagate back to us? Do we need to ensure we check the return of flags.parse?

Reply:
I believe that it is handled the same way as if you were to using flags.IntVar() and the user provides a non-integer value for the flag. flags.Parse() will detect it, print the defaults out and os.Exit(2) according to the documentation. We do get an opportunity to provide an error message so that we can tell the user what was wrong with their value. In this case it complains that the format was already set.

Reply:
We don't need to check the return of flags.Parse() because we have chosen a flags policy that does an os.Exit(2) before flags.Parse() returns.